### DISCUSSION TITLE
.required({requiredProperty: true}) expects 0 arguments, but got 1

### DISCUSSION THREAD

**Discussion by LowArmour** [colinhacks/zod#1881] [Q&A] [answered]
```ts
const optionalObject = z
    .object({
        name: z.string({ required_error: r.name.required }).min(3, r.name.min).max(20, r.name.max).trim(),
        email: z.string({ required_error: r.email.required }).email(r.email.email).trim(),
        confEmail: z.string({ required_error: r.email.required }).email(r.email.email).trim(),
        request: z.string({ required_error: r.request.required }).min(3, r.request.min).trim(),
        acceptedTerms: z.literal(true, { required_error: r.acceptedTerms.oneOf }),
    })
    .partial()
const requiredObject = optionalObject.required({name: true})
```
When calling optionalObject.required({name: true}), typescript complains that it's expecting 0 arguments, but got 1.
Indeed, when checking the source code, there is only required().
But the docs are saying that:
"You can also specify which properties to make required:

```ts
const requiredEmail = user.required({
  email: true,
});
```
"
How can I make only a property required from the whole schema which was turned into optional by .partial()?

**Comment by p10ns11y** [comment]
It seems it will be fixed in new release (https://github.com/colinhacks/zod/blob/master/src/types.ts#L1931-L1975). The code change is already there (https://www.diffchecker.com/NqoWcUu7). The current version don't support partial selection of props (https://github.com/colinhacks/zod/blob/v3.19.1/src/types.ts#L1930-L1949)

**Comment by p10ns11y** [comment]
@LowArmour you can do a patch using https://www.npmjs.com/package/patch-package

**Comment by dnishiyama** [comment]
Hey @p10ns11y thanks for the recommendation to patch this package to enable masks for `required`. I tried to follow your instructions, but you have provided a types.ts file and my node modules has types.js and types.d.ts. Is that a configuration option I have enabled? Just wondering how I can apply your change onto my `node_modules/zod/lib/types.js` and `node_modules/zod/lib/types.d.ts` files

**Comment by RSS1102** [comment]
Any recent news? About this question

**Comment by JacobWeisenburger** [comment, marked as answer]
Everything seems to be working fine for me. 

```ts
const optionalObject = z.object( { name: z.string() } ).partial()
const requiredObject = optionalObject.required( { name: true } )

console.log( requiredObject.safeParse( { name: 'foo' } ).success ) // true
console.log( requiredObject.safeParse( {} ).success ) // false
```

If this is still a problem, I recommend updating to the latest version of zod.
